src/tests/CMakeLists.txt

```diff
@@ -1,6 +1,9 @@
 set(TESTS unit-tests.cpp)
+set(SOURCES ../app/doctor/doctor.cpp ../app/person/person.cpp ../app/visit/visit.cpp)
+set(CMAKE_OSX_SYSROOT "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk")
 
-add_executable(${PROJECT_NAME}-ut ${TESTS})
+
+add_executable(${PROJECT_NAME}-ut ${TESTS} ${SOURCES})
```

Comment:
There will be no need for sources if we link libraries of classes made separately. (please refer to my comment for src/tests/CMakeLists.txt)